src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSource.cs

```diff
@@ -37,6 +37,12 @@ public abstract partial class DiagnosticSource
         [RequiresUnreferencedCode(WriteRequiresUnreferencedCode)]
         public abstract void Write(string name, object? value);
 
+        /// <inheritdoc cref="Write"/>
+        /// <typeparam name="T">The type of the value being passed as a payload for the event.</typeparam>
+        [RequiresUnreferencedCode(WriteRequiresUnreferencedCode)]
+        public void Write<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicProperties)] T>(string name, T? value) =>
```

Comment:
> `T? value`

What's the benefit of the `?` here?  If someone wants to pass in a nullable reference type, their `T` itself can be nullable, and that'll be inferred if the type of the expression being passed in is maybe null.

Reply:
This was just me not thinking and blindly copying the `object? value` overload and changing `object` to `T`. I've removed the `?`.